### DISCUSSION TITLE
Solax Solar Yield vs. Total Yield

### DISCUSSION THREAD

**Discussion by salzrat** [wills106/homeassistant-solax-modbus#936] [General]
I have a Solax X3-Hybrid G4, running latest firmware and also latest version of the Solax modbus plugin.

I wonder about the sensor.solax_today_s_solar_energy. It does not seem to correspond to the solax_today_s_yield sensor sensor shown even when taking battery into account (i.e., reducing it by battery_in-battery_out). 

For example, yesterday I had 52.6kwh today solar energy, 6.9kwh battery input, and 6.3kwh battery output. That means 52kwh should be available from the inverter, but the today total yield only gives 50.6kwh, so 1.4kwh are missing.

My assumption is that the solar energy is somehow at the input side of the inverter, and that the 1.4kwh are the inverter "loss". Is that correct? Wouldn't it then be more correct to use the "today yield" + battery_in-battery_out as the solar panel production? Because Homeassistant also calculates the house consumption from this, and this then (to me incorrectly) includes the inverter loss...

thanks!

**Comment by wills106** [comment]
We have tried a few different ways to include loses such as `sensor.solax_house_load_alt` which fudges the house load to remove losses.
But that generates it's own problems, if you have a really strong MPPT string that would skew the reading.

So all of the sensors just report what the Inverter tells the Integration.

It's down to the user to come up with any template sensors they wish. The Integration only returns what the Inverter responds with.

**Reply by salzrat** [reply]
Thanks, I understand. I guess you are right, also the battery is involved in the losses as it's still in the DC part of the pipeline, so adjusting by battery may not be accurate as well. I'll just add a template sensor for the adjusted solar yield to be able to compare quickly, and maybe add an inverter loss template that I can use as a separate energy consumer...
